staging/src/k8s.io/dynamic-resource-allocation/deviceclass/cache/cache.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"fmt"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	resourceapi "k8s.io/api/resource/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	resourceinformers "k8s.io/client-go/informers/resource/v1"
+	"k8s.io/client-go/tools/cache"
+	klog "k8s.io/klog/v2"
+)
+
+// ExtendedResourceCache maintains a global cache of extended resource to device class mappings
+// that can also be accessed by event handlers without requiring an extra scheduling cycle.
```

Comment:
"can be accessed by event handlers without requiring an extra scheduling cycle" sounds rather odd. Why should access need an extra cycle?

Here's an attempt to describe this thing:

```suggestion
// ExtendedResourceCache caches the mapping from extended resource name to
// device class. The mapping is updated based on informer events.
```